src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -161,11 +150,11 @@ private void retryEventsOnLoop(String userId, String deviceId) {
                   }
                   if (shouldReduceEventCount) {
                     if (!isLastTry) {
-                      eventCount /= 2;
```

Comment:
Why'd you move this down?

Reply:
I realized the original way is not correct. Basically eventsBuffer is not changed in the loop or by subList. The original way we just call more event callbacks then we want to.